server/Lecture.ts

```diff
@@ -1,31 +1,103 @@
 import { v4 } from "https://deno.land/std/uuid/mod.ts";
 import { cryptoRandomString } from "https://deno.land/x/crypto_random_string@1.0.0/mod.ts";
+import { WebSocketClient } from "https://deno.land/x/websocket@v0.1.1/mod.ts";
+import { Payload, QuizRequestPayload, ServerQuizRequestPayload, ServerQuizResponsePayload, QuizEndedPayload } from "./@types/payloads/types.d.ts";
+import Quiz from "./Quiz.ts";
+import Student from "./Student.ts";
 import StudentList from "./StudentList.ts";
-import { WebSocketClient} from "https://deno.land/x/websocket@v0.1.1/mod.ts";
 
 class Lecture {
     tutor: string;
     id: string;
     link: string;
     studentList: StudentList;
     wsc?: WebSocketClient;
+    quizes: Map<String, Quiz>;
 
     constructor(tutor: string) {
         this.tutor = tutor;
         this.id = v4.generate();
         this.link = cryptoRandomString({ length: 7, type: "numeric" });
         this.studentList = new StudentList();
+        this.quizes = new Map;
     }
 
-    setWebSocketClient(wsc: WebSocketClient): void{
+    setWebSocketClient(wsc: WebSocketClient): void {
         this.wsc = wsc;
         this.studentList.on("studentAdded", () => {
-            this.wsc?.send("studentAdded");        
+            this.wsc?.send("studentAdded");
         });
-        this.studentList.on("studentDeleted", () =>{
+        this.studentList.on("studentDeleted", () => {
```

Comment:
does "studentDeleted" represent the state when the connection between the student and the app is broken?

Reply:
it`s from previous version of code, but studentDeleted is emitted when student is deleted from lists of students in lecture